src/compiler/tree.cc

```diff
@@ -332,8 +423,23 @@ void TreeGrower::grow(Program* program) {
             method_queue.push_back(probe);
           }
         }
+        found_filtered_selectors.for_each([&](CallSelector selector, UnorderedSet<Method*>& methods) {
+          if (methods.empty()) return;
```

Comment:
This code is repeated. Maybe factor it out by adding an abstraction around the filtered selectors?

Reply:
changed.